### PR TITLE
[C11] Diagnose C11 keywords as being incompatible w/earlier standards

### PR DESCRIPTION
**Pull request by AaronBallman**
Our usual pattern when issuing an extension warning is to also issue a
default-off diagnostic about the keywords not being compatible with
standards before a certain point. This adds those diagnostics for C11
keywords.

### CODE REVIEW

File: clang/lib/Parse/Parser.cpp [hunk 1]
```diff
@@ -2742,6 +2742,19 @@ bool Parser::parseMisplacedModuleImport() {
   return false;
 }
 
+void Parser::diagnoseUseOfC11Keyword(const Token& Tok) {
+  // Warn that this is a C11 extension if in an older mode or if in C++.
+  // Otherwise, warn that it is incompatible with standards before C11 if in
+  // C11 or later.
+  unsigned DiagId;
+  if (!getLangOpts().C11) {
```
Comment (erichkeane) [inline clang/lib/Parse/Parser.cpp:2750]: Don't use curleys on 1 liners

File: clang/lib/Parse/Parser.cpp [hunk 2]
```diff
@@ -2742,6 +2742,19 @@ bool Parser::parseMisplacedModuleImport() {
...
+    DiagId = diag::ext_c11_feature;
+  } else {
+    DiagId = diag::warn_c11_compat_keyword;
+  }
+  Diag(Tok, DiagId) << Tok.getName();
```
Comment (erichkeane) [inline clang/lib/Parse/Parser.cpp:2755]: I dont think it is worth the DiagId pattern here, I think just a Diag + Ternary is probably fine.

File: clang/lib/Parse/Parser.cpp (hunk 1 above)
Comment (AaronBallman) [reply clang/lib/Parse/Parser.cpp]: Done

File: clang/lib/Parse/Parser.cpp (hunk 2 above)
Comment (AaronBallman) [reply clang/lib/Parse/Parser.cpp]: Done

**Review by AaronBallman — COMMENTED** [review]
Made both changes (thanks) and moved the test file to a more appropriate location.

**Review by erichkeane — APPROVED** [review]

**Pull request merged**